### ISSUE TITLE
Error calculating clear ir

### ISSUE THREAD

**Issue by scotthavens** [USDA-ARS-NWRC/smrf#16]
In gitlab by @micahjohnson150 on Jul 10, 2017, 14:06

Tuol run in water year 2015 crashes on first time step on 
DEBUG:smrf.distribute.solar:Calculating clear sky radiation, ir

Here is the error:[nohup.out](/uploads/95510c0e8e8841144ad6bf5516ad9a13/nohup.out)[log.out](/uploads/2589c5f0086f6e7867c99ddc430f8eb9/log.out)
Traceback (most recent call last):
  File "runsmrf.py", line 59, in <module>
    s.distributeData()
  File "/home/public/virturalenv/smrf/local/lib/python2.7/site-packages/smrf/framework/model_framework.py", line 361, in distributeData
    self.distributeData_single()
  File "/home/public/virturalenv/smrf/local/lib/python2.7/site-packages/smrf/framework/model_framework.py", line 438, in distributeData_single
    self.distribute['albedo'].albedo_ir)
  File "/home/public/virturalenv/smrf/local/lib/python2.7/site-packages/smrf/distribute/solar.py", line 296, in distribute
    val_beam, val_diffuse = self.calc_ir(min_storm_day, wy_day, tz_min_west, wyear, cosz, azimuth)
  File "/home/public/virturalenv/smrf/local/lib/python2.7/site-packages/smrf/distribute/solar.py", line 547, in calc_ir
    ir = ipw.IPW(self.ir_file)
  File "/home/public/virturalenv/smrf/local/lib/python2.7/site-packages/smrf/ipw/ipw.py", line 181, in __init__
    self.read(fname)
  File "/home/public/virturalenv/smrf/local/lib/python2.7/site-packages/smrf/ipw/ipw.py", line 220, in read
    raise Exception('Unexpectedly reached EOF')
Exception: Unexpectedly reached EOF

@Shavens could you elaborate on this? I looked at the code and you wrote:

```
            # fail safe, haven't needed it, but if this is running
            # on a server not a bad idea to have it here
            if tell() == st_size:
                raise Exception('Unexpectedly reached EOF')
```
Any reason why you were concerned about this on the server?

**Comment by scotthavens** [comment]
In gitlab by @micahjohnson150 on Jul 10, 2017, 14:08

Note that this was ran using a branch of smrf at commit 3f852d07 so it does not include any of the newer stuff you put for thermal or the stuff I have put in for snow density

**Comment by scotthavens** [comment]
In gitlab by @shavens on Jul 10, 2017, 14:29

Most of that code was adapted from another project so I don't believe that was one of my additions. But it could have been I guess.

It's trying to read the IPW image and reaching the end of the file. So there is a problem with the IPW file. You should be able to see it in the working dir and look at it from the command line using the IPW utilities. From the `ialbedo` error

```
ialbedo: ERROR:
	day, 0, before start, 426.624
```
This is really the error you'll want to track down.

**Comment by scotthavens** [comment]
In gitlab by @micahjohnson150 on Jul 11, 2017, 15:02

I think I have also resolved this issue. A while back all the files' permissions were changed on the server. I not sure this is what cause the above problem but after a reinstall of IPW it appears to not be a problem.

I am thinking @shavens we should probably move to a smrf contained version of IPW so that we can have a little more control on the IPW beast. We can git track it still in its own repo but have a copy that's pointed to a specific commit in smrf.
